Bug-Fix: Unable to merge clips from two search results. Issue #123

varunchopra-1 opened the pull request:
- A new video_data list is added that keeps track of each video clip (shot) that matches the search.
- When a search succeeds, each matching video clip now includes its source video ID, start/end times, title, and a relevance score showing how well it matches the search query.

Issue: #123 

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- **Improvements**
	- Refined search functionality with more efficient result handling.
	- Enhanced error management for search operations, particularly for the "spoken_word" index type.
	- Improved data collection and summary generation for search results.
	- Simplified agent description and search logic.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

ankit-v2-3 commented on backend/director/agents/search.py:
```diff
@@ -60,7 +60,7 @@
     def __init__(self, session: Session, **kwargs):
         self.agent_name = "search"
-        self.description = "Agent to search information from VideoDB collections. Mainly used with a collection of videos."
+        self.description = "Agent to search information from VideoDB collections."
```
let's revert this description change.

ankit-v2-3 commented on backend/director/agents/search.py:
```diff
@@ -153,39 +154,40 @@ def run(
                 )
+
             search_result_videos = {}
+            video_data = []
```
Let's remove `video_data`, `search_result_videos` is enough.

varunchopra-1 replied:
Also from AgentResponse() at line 247?
I mean do we have to return videos data in response seperately as in line 247?

If yes we can remove `video_data = []` and fetch video data for response from search_result_videos.

ankit-v2-3 replied:
yes, remove `video_data` from response

ankit-v2-3 reviewed (changes requested):
Submitted the review.

varunchopra-1 commented:
Added error handling for missing spoken word indexes in video search.
- Implemented `get_transcript` to check if the spoken word index exists before performing a search because the search method does not raise an error when the index is missing; it just returns an empty list.
- Now raises an error when the video is not indexed instead of returning an empty list.

ankit-v2-3 reviewed (approved):
looks good to me.
